src/transform-layer.coffee

```diff
@@ -64,52 +69,49 @@ class TransformLayerIterator
     else
       {value: undefined, done: true}
 
-  seek: (position, clip=CLIP_BACKWARD) ->
+  seek: (position, clip) ->
+    calculateOvershoot = (position, lastPosition, lastSourcePosition) =>
+      overshoot = position.traversalFrom(lastPosition)
+      lastSourcePosition = lastSourcePosition.traverse(overshoot)
+      @position = position
+      @sourcePosition = lastSourcePosition
+
+    @_seek (=> @position), position, calculateOvershoot, clip
+
+  seekToSourcePosition: (position, clip) ->
+    calculateOvershoot = (position, lastPosition, lastSourcePosition) =>
+      overshoot = position.traversalFrom(lastSourcePosition)
+      lastPosition = lastPosition.traverse(overshoot)
+      @position = lastPosition
+      @sourcePosition = position
+
+    @_seek (=> @sourcePosition), position, calculateOvershoot, clip
+
+  _seek: (seekPropertyFn, position, calculateOvershoot, clip = CLIP_BACKWARD) ->
```

Comment:
We think we should just keep the duplication for now. Another approach might be to extract a method for performing the clipping and/or the overshoot calculation. For now though, while things are still taking shape, I think the duplication is ok.